tests/wpunit/PostObjectConnectionQueriesTest.php

```diff
@@ -732,12 +732,12 @@ public function testPostExcerptsAreDifferent() {
 
 		$post_1_args = [
 			'post_content' => 'Post content 1',
-			'post_excerpt' => 'Post excerpt 1',
+			'post_excerpt' => '',
 		];
 
 		$post_2_args = [
 			'post_content' => 'Post content 2',
-			'post_excerpt' => 'Post excerpt 2',
+			'post_excerpt' => '',
```

Comment:
I feel like we should leave the explicitly set `post_excerpt` arg here so we don't get some false positives on this working when it's actually not. The best way to test the global $post data is being setup correctly is by testing the content & excerpt data is actually different. With this change the excerpts are the same for both of the posts we're querying.

Reply:
What I found was exactly the opposite. This test was passing when it should have failed. When excerpt has a value it does not need to generate the excerpt as it can return the value directly. In that case the `get_the_excerpt` filter here:

https://github.com/wp-graphql/wp-graphql/blob/c9ca28a1d1bdcf34a773220e28412b401dc4e907/src/Type/Object/PostObject.php#L265

does not use the global post for anything because the default filter in WP uses the `wp_trim_excerpt` function:

https://github.com/WordPress/WordPress/blob/d9a1f99d9ce352a0e151169c3612b00c4c146ddf/wp-includes/default-filters.php#L185


And the `wp_trim_excerpt` function checks for empty string here and bails out if there is something already:

https://github.com/WordPress/WordPress/blob/cddee8b72e59c9968b1c786c01992dae6f71ee13/wp-includes/formatting.php#L3685-L3712

Reply:
But we could probably write a completely new and better test for this situation.

Fox example we could just assert that when querying multiple posts the global post is set to it inside the `graphql_resolve_field` filter.

Reply:
Ahhhhh, I see. So in this test, the post_excerpt should be the same as the post_content for each, shouldn't it?

Reply:
Yeah, if the excerpt field is empty it will be generated from the content field on demand so it should be always different in this test.

Reply:
I'll rebase my commits a bit. Will ping you when I think this is ready to merge.